### PR TITLE
[guilib][vfs][imagecache][pvr] Load PVR group images into texture cache when viewed, like standard images

### PR DESCRIPTION
**Pull request by rmrector**
## Description
Use the new interface from #23197 to load generated PVR group thumbs into the texture cache.

This is the last special image type currently in the texture cache to be converted.

## Motivation and context
Consistent behavior:

- all images load the same way, whether they are already cached or not.
- Remote interfaces can show these images even if they haven't been viewed in Kodi GUI yet.
- Kodi doesn't have to pre-cache a potentially massive list of images when browsing folders, just in case they are viewed, saving both storage size in the cache and taking queue time for images actually displayed in the GUI.
- These special images can now be preloaded the same as other images.

Easier maintenance and development: this provides a clear interface to load more special images, which avoids the complication of designing and developing a way to preload some types of artwork.

## How has this been tested?
Real quick on Windows with the demo PVR add-on. I don't use PVR so I can't test out during regular usage.

## What is the effect on users?
Minimal I think. Small changes around the edges related to the consistent behavior motivation.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/PVRThumbLoader.cpp [hunk 1]
```diff
@@ -102,35 +99,9 @@ bool CPVRThumbLoader::FillThumb(CFileItem& item)
...
-      CServiceBroker::GetTextureCache()->AddCachedTexture(thumb, details);
-      return thumb;
-    }
-  }
+  std::string thumb = StringUtils::Format(
+      "{}?ts={}", // append timestamp to Thumb URL to enforce texture refresh
+      CTextureUtils::GetWrappedImageURL(channelGroupItem.GetPath(), "pvr"), std::time(nullptr));
 
-  return {};
+  return thumb;
```
Comment (ksooo) [inline xbmc/pvr/PVRThumbLoader.cpp:106]: Maybe we can get rid off the variable here and simply return the stringutils::format result.

File: xbmc/pvr/PVRChannelGroupImageFileLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  {
+    const std::string& icon = channel->GetArt("icon");
+    if (!icon.empty())
+      channelIcons.emplace_back(CPVRCachedImages::UnwrapImageURL(icon));
+
+    if (channelIcons.size() == 9) // limit number of tiles
+      break;
+  }
+
+  return CPicture::CreateTiledThumb(channelIcons);
```
Comment (ksooo) [inline xbmc/pvr/PVRChannelGroupImageFileLoader.cpp:46]: The old implementation did much more after calling `CPicture::CreateTiledThumb`. Why is this no longer needed? Just asking.

```c++
    if (CPicture::CreateTiledThumb(channelIcons, CTextureCache::GetCachedPath(relativeCacheFile)))
    {
      CTextureDetails details;
      details.file = relativeCacheFile;
      details.width = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_imageRes;
      details.height = details.width;
      CServiceBroker::GetTextureCache()->AddCachedTexture(thumb, details);
      return thumb;
    }
```

**Review by ksooo — COMMENTED** [review]
Did no runtime-test, code change looks good.

File: xbmc/pvr/PVRChannelGroupImageFileLoader.cpp (hunk 2 above)
Comment (rmrector) [reply xbmc/pvr/PVRChannelGroupImageFileLoader.cpp:46]: "Motivation: Consistent behavior" That code just manually saved one particular image type to the texture cache. Now all image types are saved to the texture cache in one place in the texture cache code after any image is loaded, no need to duplicate it.

File: xbmc/pvr/PVRThumbLoader.cpp (hunk 1 above)
Comment (rmrector) [reply xbmc/pvr/PVRThumbLoader.cpp]: updated

**Review by ksooo — APPROVED** [review]
Your green button, @rmrector

**Comment by rmrector** [comment]
thanks

**Pull request merged**